dotnet/src/SemanticKernel.Abstractions/Contents/BinaryContent.cs

```diff
@@ -3,118 +3,310 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
-using System.IO;
+using System.Text;
 using System.Text.Json.Serialization;
-using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Text;
+
+#pragma warning disable CA1056 // URI-like properties should not be strings
+#pragma warning disable CA1055 // URI-like parameters should not be strings
+#pragma warning disable CA1054 // URI-like parameters should not be strings
 
 namespace Microsoft.SemanticKernel;
 
 /// <summary>
 /// Provides access to binary content.
 /// </summary>
-[Experimental("SKEXP0010")]
+[Experimental("SKEXP0001")]
 public class BinaryContent : KernelContent
 {
-    private readonly Func<Task<Stream>>? _streamProvider;
+    private string? _dataUri;
+    private ReadOnlyMemory<byte>? _data;
+    private Uri? _referencedUri;
 
     /// <summary>
     /// The binary content.
     /// </summary>
-    public ReadOnlyMemory<byte>? Content { get; set; }
+    [JsonIgnore, Obsolete("Use Data instead")]
+    public ReadOnlyMemory<byte>? Content => this.Data;
 
     /// <summary>
-    /// Initializes a new instance of the <see cref="BinaryContent"/> class.
+    /// Gets the referenced Uri of the content.
     /// </summary>
```

Comment:
```suggestion
    /// </summary>
    /// <remarks>
    /// This is a uri that points at the location of the content if it is available for reading from an external source.
    /// </remarks>```

Reply:
I will keep the description simple, as this Uri can be there just as informative or extra, not necessarily implying the caller will do something specific with it.